Gems/Atom/RPI/Code/Source/RPI.Reflect/Material/MaterialAsset.cpp

```diff
@@ -146,6 +152,30 @@ namespace AZ
             }
         }
         
+        void MaterialAsset::RealignPropertyValuesAndNames()
+        {
+            const MaterialPropertiesLayout* propertyLayout = GetMaterialPropertiesLayout();
+            AZStd::vector<MaterialPropertyValue> alignedPropertyValues(m_materialTypeAsset->GetDefaultPropertyValues().begin(), m_materialTypeAsset->GetDefaultPropertyValues().end());
+            for (size_t i = 0; i < m_propertyNames.size(); ++i)
+            {
+                const MaterialPropertyIndex propertyIndex = propertyLayout->FindPropertyIndex(m_propertyNames[i]);
+                if (propertyIndex.IsValid())
```

Comment:
Let's report a warning when this is false.

Reply:
Added warning and tested as well.